### DISCUSSION TITLE
Few Questions on the Usage of Shared Preset

### DISCUSSION THREAD

**Discussion by juicybaba** [renovatebot/renovate#21257] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I am working on GitHub org level preset by using `.github` + `renovate-config.json`
1. Is it possible to onboard renovate to all repo without having `renovate.json`?
2. If 1 is not doable, as per https://docs.renovatebot.com/config-presets/#organization-level-presets, is below config the minimum content for `renovate.json` in each repo?
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["local>myorgname/.github:renovate-config"]
}
```
3. When working on `renovate-config.json`, should I use `Included Presets` (e.g. `default`/`schedule` which are used under `extends`) or [Repository Configuration](https://docs.renovatebot.com/configuration-options/) (used under the root of `renovate-config.json`) or both?

### What you tried so far.

1. never worked
2. I am looking for an answer
3. I am looking for an answer, there is an overlap in capability perspective, and each has some unique capability. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
> 1. Is it possible to onboard renovate to all repo without having `renovate.json`?

Currently this is possible if myself or another Mend staff can update your org settings manually. It sounds like you want `onboarding=false`, `requireConfig=optional`? Please confirm if so, as well as which org. It may take us up to 2-3 days to apply the setting.

> 3. When working on `renovate-config.json`, should I use `Included Presets` (e.g. `default`/`schedule` which are used under `extends`) or [Repository Configuration](https://docs.renovatebot.com/configuration-options/) (used under the root of `renovate-config.json`) or both?

I'm not sure I understand your question, but your global config can both include other presets (especially the built-in ones) plus regular repository config.

**Reply by juicybaba** [reply]
> > 1. Is it possible to onboard renovate to all repo without having `renovate.json`?
> 
> Currently this is possible if myself or another Mend staff can update your org settings manually. It sounds like you want `onboarding=false`, `requireConfig=optional`? Please confirm if so, as well as which org. It may take us up to 2-3 days to apply the setting.
> 
> > 3. When working on `renovate-config.json`, should I use `Included Presets` (e.g. `default`/`schedule` which are used under `extends`) or [Repository Configuration](https://docs.renovatebot.com/configuration-options/) (used under the root of `renovate-config.json`) or both?
> 
> I'm not sure I understand your question, but your global config can both include other presets (especially the built-in ones) plus regular repository config.

Hi @rarkins, 

Thanks for the reply. What I am trying to figure out is that is it possible to enable/onboard renovate to all repos under an org without having `renovate.json` in each repo, and manage the renovate behavior via `renovate-config.json` in `.github` repo.

And to the other question, thanks, the reason I was asking is that presets and repository config have few very similar or same config, and I wasn't able to understand which one should I use or recommended from renovate.

**Comment by juicybaba** [comment]
> > 1. Is it possible to onboard renovate to all repo without having `renovate.json`?
> 
> 
> 
> Currently this is possible if myself or another Mend staff can update your org settings manually. It sounds like you want `onboarding=false`, `requireConfig=optional`? Please confirm if so, as well as which org. It may take us up to 2-3 days to apply the setting.
> 
> 
> 
> > 3. When working on `renovate-config.json`, should I use `Included Presets` (e.g. `default`/`schedule` which are used under `extends`) or [Repository Configuration](https://docs.renovatebot.com/configuration-options/) (used under the root of `renovate-config.json`) or both?
> 
> 
> 
> I'm not sure I understand your question, but your global config can both include other presets (especially the built-in ones) plus regular repository config.
> 
> 

@rarkins Can you do it for TeckResourcesCloudInfrastructurePrivate? There will be more if everything works fine. Thanks.

**Reply by rarkins** [reply]
Please confirm exactly what you mean by "do it", so we have no misunderstanding. You have included my entire comment in your reply, so it's unclear

**Reply by juicybaba** [reply]
@rarkins 
Here is what I want to have for TeckResourcesCloudInfrastructurePrivate github org.
There is a `renovate-config.json` in `.github` under TeckResourcesCloudInfrastructurePrivate org, and everything works well when having `renovate.json` in a repo. I want to have renovate onboarded to all repos under that org without having `renovate.json` in repo.

Hope this is clear 🙂

**Reply by rarkins** [reply]
Sorry, still not clear. I assume you don't mean _any_ `renovate.json` in the repo - you have a specific content in mind - please state it

**Reply by juicybaba** [reply]
Hmm, for the setup that i have -  renovate has the permission to access all repos, and org level `renovate-config.json` is in place, is it possible for renovate to scan a repo without that repo having `renovate.json` file? Because we want to track the dependency without dumping (even a PR will be created to facilitate that) file to each repo.

**Reply by rarkins** [reply]
We can set onboarding=false, and requireConfig=optional, and that will mean Renovate runs on every repo without onboarding. But if you want each repo to extend a preset or anything like that then you need to be more specific. According to what documentation did you create a `renovate-config.json` file? I do not recall that file name being an official/default one.

**Reply by juicybaba** [reply]
I have created a `renovate-config.json` file under the `.github` repo. And if anyone wants extra flavor, I will ask the code owner to add `renovate.json` to their repo as needed. But from the get-go, we would love to enable renovate without having `renovate.json` in each repo at all. 

Overall, your suggestion sounds good to me, pleases do it against `TeckResourcesCloudInfrastructurePrivate` org. I will do some tests with this org before asking you to do the same thing with the other org.

**Reply by rarkins** [reply]
I think what you mean is that you want this default config for each repo in your org?

```json
{
  "extends": ["local>TeckResourcesCloudInfrastructurePrivate/.github:renovate-config"],
  "onboarding": false,
  "requireConfig": "optional"
}
```

**Reply by juicybaba** [reply]
> I think what you mean is that you want this default config for each repo in your org?
> 
> ```json
> {
>   "extends": ["local>TeckResourcesCloudInfrastructurePrivate/.github:renovate-config"],
>   "onboarding": false,
>   "requireConfig": "optional"
> }
> ```

yes, the setting looks good, is this something that can be done on renovate team side for the entire org? Again, the goal is to *** not*** have `renovate.json` in each repo.

**Reply by rarkins** [reply]
I have added this config to your org manually but note these things:
1. You already have quite a lot of repos already installed, so their behaviour might now change
2. ~~You do not seem to have installed the `TeckResourcesCloudInfrastructurePrivate/.github` repo itself, which means Renovate can't access it, and all repos will fail to look it up~~

Edit: I found the repo now, never mind

**Reply by juicybaba** [reply]
> I have added this config to your org manually but note these things:
> 
> 1. You already have quite a lot of repos already installed, so their behaviour might now change
> 2. ~You do not seem to have installed the `TeckResourcesCloudInfrastructurePrivate/.github` repo itself, which means Renovate can't access it, and all repos will fail to look it up~
> 
> Edit: I found the repo now, never mind

It is fine for this org. Correct me if I am wrong, but in the future, I can still have `renovate.json` in the certain repo to overwrite or expand the org-level behavior.

**Reply by rarkins** [reply]
Yes

**Reply by juicybaba** [reply]
> Yes

awesome, please proceed and let me know once it is done. thanks. 👍

**Reply by rarkins** [reply]
It's done. Please test out on one repo at a time at first if possible, just in case I made a mistake

**Reply by juicybaba** [reply]
> It's done. Please test out on one repo at a time at first if possible, just in case I made a mistake

Awesome! In terms of testing.
1. Should I wait for a few days before testing?
2. Will the below dependency dashboard pop up automatically for a repo or there has to be some real code (Terraform in my case) in the repo?
![image](https://user-images.githubusercontent.com/79539850/234635810-6f838bbe-392d-440d-bdcf-ecad03fe3568.png)

**Reply by rarkins** [reply]
1. No
2. Repos without detected dependencies will be skipped

**Reply by juicybaba** [reply]
> 1. No
> 2. Repos without detected dependencies will be skipped

I did some tests, the dependency dashboard pop up immediately for new repo :thumbsup:, but not for the existing repo :worried:

**Reply by rarkins** [reply]
Did it run on the existing repo?

**Reply by juicybaba** [reply]
> Did it run on the existing repo?

no, it didn't, nothing happened on the existing repo (without `renovate.json` in the repo)

**Reply by rarkins** [reply]
Have you checked the logs?

**Reply by juicybaba** [reply]
> Have you checked the logs?
if this is the correct place to check the log, I don't see any log.
![image](https://user-images.githubusercontent.com/79539850/234954330-621954e4-0d16-4bd7-91ee-46da75e97b13.png)

**Reply by rarkins** [reply]
That repo had no dependencies found, so was marked as disabled. If you have repos like that, you'll need to uninstall/reinstall them to trigger a new run, otherwise they're only rechecked ~monthly

**Reply by juicybaba** [reply]
> That repo had no dependencies found, so was marked as disabled. If you have repos like that, you'll need to uninstall/reinstall them to trigger a new run, otherwise they're only rechecked ~monthly

aha, not sure if you did anything in the backend but the dependency dashboard show up, and I didn't make any chance after this [post](https://github.com/renovatebot/renovate/discussions/21257#discussioncomment-5744764) 
![image](https://user-images.githubusercontent.com/79539850/234967018-44e6f253-7ce2-4aa3-b406-70041071be69.png)
![image](https://user-images.githubusercontent.com/79539850/234970087-81159542-9804-419e-93eb-21092b2fc9bd.png)

**Reply by rarkins** [reply]
I fixed it manually

**Reply by juicybaba** [reply]
> That repo had no dependencies found

thanks, the dashboard shows there are dependencies in the code tho.

**Reply by juicybaba** [reply]
so, any idea about why that repo has the dependencies in the code but the dashboard didn't show up?

**Reply by rarkins** [reply]
Because it was marked as disabled in the backend so never retried

**Reply by juicybaba** [reply]
ok, I would like to have same setup for another org that has hundreds of repos, would you be able to help if the same thing happens to a dozen of repos? if this is do-able on your side, please do the same thing for `TeckResourcesTDS`. Thanks 😄

**Reply by rarkins** [reply]
Do you mean this?

```
{
  "extends": ["local>TeckResourcesTDS/.github:renovate-config"],
  "onboarding": false,
  "requireConfig": "optional"
}
```

**Reply by juicybaba** [reply]
> Do you mean this?
> 
> ```
> {
>   "extends": ["local>TeckResourcesTDS/.github:renovate-config"],
>   "onboarding": false,
>   "requireConfig": "optional"
> }
> ```

Yes, whatever you did for `TeckResourcesCloudInfrastructurePrivate`, do the same thing for `TeckResourcesTDS`. And I hope you don't have to enable `renovate` for existing repos individually this time, but let's see. 😄

**Reply by rarkins** [reply]
The question is, do you have a shared config for `TeckResourcesTDS` already? It can't be _exactly_ the same, because `TeckResourcesCloudInfrastructurePrivate` is private and unaccessible by `TeckResourcesTDS` repos.

**Reply by juicybaba** [reply]
> The question is, do you have a shared config for `TeckResourcesTDS` already? It can't be _exactly_ the same, because `TeckResourcesCloudInfrastructurePrivate` is private and unaccessible by `TeckResourcesTDS` repos.

Yes, there is a shared config in '.github' repo under 'TeckResourcesTDS' org.

**Reply by rarkins** [reply]
Now set for this org

**Reply by juicybaba** [reply]
Great, thanks. 
Like last time, the new repo has `renovate` enabled/onboarded automatically, but not for the existing ones. Is there a quick way to enable `renovate` in bulk as there might be hundreds repo.

**Reply by rarkins** [reply]
There were 321 repos in the org with status=disabled. I've reset them all to status=null so that they'll be retried over the next few hours

**Reply by juicybaba** [reply]
Awesome, I will keep an eye on them and let you know.

**Reply by juicybaba** [reply]
Quick question, there are a number of repos have official renovate onboarding PR open. (they never merge it, this is why we want to onboard renovate with zero-touch). Will this thing prevent renovate to be onboarded/enabled without `renovate.json`? Or I should just wait?

I can see some repo have renovate enabled automatically, but not all of repos, trying to figure out why.

![image](https://user-images.githubusercontent.com/79539850/236568053-45f3a888-c3b1-48fa-88ea-b8d991a99e22.png)

**Reply by rarkins** [reply]
Your screenshot is of a dashboard issue, not an onboarding PR

**Reply by juicybaba** [reply]
> Your screenshot is of a dashboard issue, not an onboarding PR

ugh, sorry, I mean this PR.
![image](https://user-images.githubusercontent.com/79539850/236878071-8e49e363-9020-4a36-bb9c-85ea613d8020.png)

**Reply by rarkins** [reply]
Those PRs will get cleared up the next time Renovate checks onboarding repos. To speed it up, they can close the PR unmerged

**Reply by juicybaba** [reply]
thanks, everything looks good so far. I would love to say renovate is an amazing tool, really appreciated the contribution from you and the team. ❤️

**Reply by juicybaba** [reply]
@rarkins Everything used to work but it looks like renovate doesn't close its dependency PR when the code is updated via other PR, I think all repo by default should use the `renovate-config.json` in `.github` repo due to our previous chat. Would you mind take a look?

PS: It works if I add `renovate.json` with proper config to repo.

Log: 
![image](https://github.com/renovatebot/renovate/assets/79539850/fdd1d9c3-798c-4993-9ad3-1b7a352f36e2)
![image](https://github.com/renovatebot/renovate/assets/79539850/36d1bfbb-6459-48f1-af5c-bec702ac2ae7)

The onboarding PR that I closed to make renovate dashboard shows up, this is also the PR in the log screenshot.
![image](https://github.com/renovatebot/renovate/assets/79539850/b9f4f093-1323-4ad4-82df-409e93b1a7cc)

PR which should be closed as dependency has been fixed: 
![image](https://github.com/renovatebot/renovate/assets/79539850/d738d4ea-f248-4299-965f-9d859e633691)

**Reply by rarkins** [reply]
So the repository is disabled due to no config, but some old PRs remain?

**Reply by juicybaba** [reply]
hmm, based on our previous [chat](https://github.com/renovatebot/renovate/discussions/21257#discussioncomment-5812094), my understanding is renovate bot should be enabled for all repo (existing and new) under `TeckResourcesCloudInfrastructurePrivate` and `TeckResourcesTDS` org without needing `renovate.json` in the repo.

The problem I am having is, renovate opened a number of PRs, and the user fixed all dependencies in one shot from a PR he made, the renovate is not able to close its own PRs, I tried checking the checkbox in renovate dashboard, but it doesn't help, log is attached in previous msg.

**Reply by rarkins** [reply]
Please attach the full logs for that repo

**Reply by juicybaba** [reply]
here you go
```
DEBUG: File config
{
  "config": {}
}

DEBUG: CLI config
{
  "config": {}
}

DEBUG: Env config
{
  "config": {
    "allowPostUpgradeCommandTemplating": true,
    "allowedPostUpgradeCommands": [
      "^git add --all$",
      "^git reset$",
      "^npx beachball change( --no-fetch)? --no-commit --type (patch|none) --message '{{{commitMessage}}}'$",
      "^pwd$"
    ],
    "baseDir": "/tmp/worker/d91dcc/cf3091",
    "dependencyDashboardFooter": "\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n",
    "extends": [
      "github>whitesource/merge-confidence:beta"
    ],
    "forkProcessing": "disabled",
    "gitAuthor": "renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>",
    "gitIgnoredAuthors": [
      "29139614+renovate[bot]@users.noreply.github.com"
    ],
    "hostRules": [
      {
        "hostType": "docker",
        "matchHost": "docker.io",
        "password": "***********",
        "username": "mdpprod2renovate"
      }
    ],
    "logContext": "14d4662437c3401cacae5eb93edbd655",
    "logFile": "/tmp/worker/d91dcc/cf3091/github/TeckResourcesTDS/terraform-azure-main-plant-run/14d4662437c3401cacae5eb93edbd655.log",
    "logFileLevel": "debug",
    "onboardingConfig": {
      "$schema": "https://docs.renovatebot.com/renovate-schema.json",
      "extends": [
        "config:base"
      ]
    },
    "onboardingNoDeps": false,
    "platform": "github",
    "platformCommit": true,
    "prHeader": "[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)",
    "repositories": [
      "TeckResourcesTDS/terraform-azure-main-plant-run"
    ],
    "token": "***********",
    "username": "renovate[bot]",
    "repositoryCache": "enabled",
    "repositoryCacheType": "s3://mend-developer-platform-prod/renovate/",
    "binarySource": "docker",
    "redisUrl": "redis://mend-developer-platform-renovate-prod.aqffol.ng.0001.use1.cache.amazonaws.com:6379",
    "dockerChildPrefix": "renovate_a_",
    "dockerImagePrefix": "ghcr.io/containerbase",
    "privateKey": "***********",
    "privateKeyOld": "***********",
    "prFooter": "This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://developer.mend.io/{{platform}}/{{repository}})."
  }
}

DEBUG: Combined config
{
  "config": {
    "allowPostUpgradeCommandTemplating": true,
    "allowedPostUpgradeCommands": [
      "^git add --all$",
      "^git reset$",
      "^npx beachball change( --no-fetch)? --no-commit --type (patch|none) --message '{{{commitMessage}}}'$",
      "^pwd$"
    ],
    "baseDir": "/tmp/worker/d91dcc/cf3091",
    "dependencyDashboardFooter": "\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n",
    "extends": [
      "github>whitesource/merge-confidence:beta"
    ],
    "forkProcessing": "disabled",
    "gitAuthor": "renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>",
    "gitIgnoredAuthors": [
      "29139614+renovate[bot]@users.noreply.github.com"
    ],
    "hostRules": [
      {
        "hostType": "docker",
        "matchHost": "docker.io",
        "password": "***********",
        "username": "mdpprod2renovate"
      }
    ],
    "logContext": "14d4662437c3401cacae5eb93edbd655",
    "logFile": "/tmp/worker/d91dcc/cf3091/github/TeckResourcesTDS/terraform-azure-main-plant-run/14d4662437c3401cacae5eb93edbd655.log",
    "logFileLevel": "debug",
    "onboardingConfig": {
      "$schema": "https://docs.renovatebot.com/renovate-schema.json",
      "extends": [
        "config:base"
      ]
    },
    "onboardingNoDeps": false,
    "platform": "github",
    "platformCommit": true,
    "prHeader": "[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)",
    "repositories": [
      "TeckResourcesTDS/terraform-azure-main-plant-run"
    ],
    "token": "***********",
    "username": "renovate[bot]",
    "repositoryCache": "enabled",
    "repositoryCacheType": "s3://mend-developer-platform-prod/renovate/",
    "binarySource": "docker",
    "redisUrl": "redis://mend-developer-platform-renovate-prod.aqffol.ng.0001.use1.cache.amazonaws.com:6379",
    "dockerChildPrefix": "renovate_a_",
    "dockerImagePrefix": "ghcr.io/containerbase",
    "privateKey": "***********",
    "privateKeyOld": "***********",
    "prFooter": "This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://developer.mend.io/{{platform}}/{{repository}})."
  }
}

DEBUG: Found valid git version: 2.40.1
DEBUG: Using default github endpoint: https://api.github.com/
DEBUG: Platform config
{
  "platformConfig": {
    "hostType": "github",
    "endpoint": "https://api.github.com/",
    "isGHApp": true,
    "isGhe": false
  }
  "renovateUsername": "renovate[bot]"
}

DEBUG: Using configured gitAuthor (renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>)
DEBUG: Adding token authentication for api.github.com to hostRules
DEBUG: Using configured baseDir: /tmp/worker/d91dcc/cf3091
DEBUG: Using cacheDir: /tmp/worker/d91dcc/cf3091/cache
DEBUG: Using containerbaseDir: /tmp/worker/d91dcc/cf3091/cache/containerbase
DEBUG: Redis cache init
DEBUG: Commits limit = null
DEBUG: Setting global hostRules
DEBUG: Adding password authentication for docker.io to hostRules
DEBUG: Adding token authentication for api.github.com to hostRules
DEBUG: validatePresets()
DEBUG: Reinitializing hostRules for repo
DEBUG: Clearing hostRules
DEBUG: Adding password authentication for docker.io to hostRules
DEBUG: Adding token authentication for api.github.com to hostRules
DEBUG: No dangling containers to remove
INFO: Repository started
{
  "renovateVersion": "35.159.7"
}

DEBUG: Using localDir: /tmp/worker/d91dcc/cf3091/repos/github/TeckResourcesTDS/terraform-azure-main-plant-run
DEBUG: PackageFiles.clear() - Package files deleted
DEBUG: initRepo("TeckResourcesTDS/terraform-azure-main-plant-run")
DEBUG: TeckResourcesTDS/terraform-azure-main-plant-run default branch = main
DEBUG: Using app token for git init
DEBUG: RepoCacheS3.read() - success
DEBUG: Repository cache is restored from revision 13
DEBUG: Resetting npmrc
DEBUG: Resetting npmrc
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: findPr(renovate/configure, Configure Renovate, !open)
DEBUG: getPrList success
{
  "pullsTotal": 18
  "requestsTotal": 14
  "apiQuotaAffected": true
}

DEBUG: Found PR #27
DEBUG: Checking cached config file name
DEBUG: GET https://api.github.com/repos/TeckResourcesTDS/terraform-azure-main-plant-run/contents/renovate.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=130)
DEBUG: Existing config file no longer exists
DEBUG: findFile(renovate.json)
DEBUG: Initializing git repository into /tmp/worker/d91dcc/cf3091/repos/github/TeckResourcesTDS/terraform-azure-main-plant-run
DEBUG: Performing blobless clone
DEBUG: git clone completed
{
  "durationMs": 708
}

DEBUG: latest repository commit
{
  "latestCommit": {
    "hash": "fca2c7b8e8d36175acc08a5ebbb5f03041cc0c80",
    "date": "2023-07-07T15:58:08-04:00",
    "message": "Merge pull request #263 from TeckResourcesTDS/feature/main-to-prod-fixes",
    "refs": "HEAD -> main, origin/main, origin/HEAD",
    "body": "Feature main to prod conflicts fixes",
    "author_name": "Alex Maistrov",
    "author_email": "maistrovas@gmail.com"
  }
}

DEBUG: findFile(renovate.json5)
DEBUG: findFile(.github/renovate.json)
DEBUG: findFile(.github/renovate.json5)
DEBUG: findFile(.gitlab/renovate.json)
DEBUG: findFile(.gitlab/renovate.json5)
DEBUG: findFile(.renovaterc)
DEBUG: findFile(.renovaterc.json)
DEBUG: findFile(.renovaterc.json5)
DEBUG: config file not found
DEBUG: Found closed onboarding PR
DEBUG: Repo is not onboarded and no merged PRs exist
DEBUG: Getting comments for #27
DEBUG: Found 1 comments
DEBUG: Ensuring comment "Renovate is disabled" in #27
DEBUG: Comment is already update-to-date
INFO: Repository is disabled - skipping
DEBUG: Repository result: disabled-closed-onboarding, status: disabled, enabled: false, onboarded: undefined
DEBUG: Repository timing splits (milliseconds)
{
  "splits": {}
  "total": 20780
}

DEBUG: Package cache statistics
{
  "get": {
    "count": 0
  }
  "set": {
    "count": 0
  }
}

DEBUG: http statistics
{
  "urls": {
    "https://api.github.com/repos/TeckResourcesTDS/terraform-azure-main-plant-run/contents/renovate.json (GET,404)": 1,
    "https://api.github.com/repos/TeckResourcesTDS/terraform-azure-main-plant-run/issues/27/comments (GET,200)": 1,
    "https://api.github.com/repos/TeckResourcesTDS/terraform-azure-main-plant-run/pulls (GET,200)": 14
  }
  "hostStats": {
    "api.github.com": {
      "requestCount": 16,
      "requestAvgMs": 1101,
      "queueAvgMs": 0
    }
  }
  "totalRequests": 16
}

DEBUG: Package lookup durations
DEBUG: dns cache
{
  "hosts": []
}

INFO: Repository finished
{
  "cloned": true
  "durationMs": 20780
}

DEBUG: Renovate exiting
```

And those are the PRs that I am expecting to be closed by renovate
<img width="1203" alt="image" src="https://github.com/renovatebot/renovate/assets/79539850/853c1b54-61b4-452f-916d-837104e97eb4">

**Reply by rarkins** [reply]
I can see the conclusion is this:

```
INFO: Repository is disabled - skipping
DEBUG: Repository result: disabled-closed-onboarding, status: disabled, enabled: false, onboarded: undefined
```

Is that part expected/correct?

**Reply by juicybaba** [reply]
hmm, my expectation was, as renovate team did something in the backend, as a result, all repo (existing and future ones) in the org should have renovate onboarded by default (using `renovate-config.json` in `.github` repo as the config file) and functioned properly without having `renovate.json` in the repo. 

The reality is, auto-onboarding did happen and dependency PR did get created as expected. However, for some reason, the dependency PRs don't get closed after the code is fixed in a user-created PR. The dependency PR will be closed automatically if a local `renovate.json` is added in the repo and points to the shared `renovate-config.json` in `.github` repo.

**Reply by rarkins** [reply]
So let's address the first point fast. Was it your expectation that your repos would skip onboarding and run _always_, regardless of whether the repo has a config file or not?

**Reply by juicybaba** [reply]
yes, that is correct.

**Reply by rarkins** [reply]
Let's fix that part first. Can you confirm which organization that is?

**Reply by juicybaba** [reply]
Can you fix both `TeckResourcesTDS` and `TeckResourcesCloudInfrastructurePrivate`?

**Reply by rarkins** [reply]
FYI @nabeelsaabna

**Reply by nabeelsaabna** [reply]
Hi @juicybaba 
only 1 of the 2 orgs has a config in the DB and it's:
```
{
	"extends": [
		"local>TeckResourcesCloudInfrastructurePrivate/.github:renovate-config"
	],
	"onboarding": false,
	"requireConfig": "optional"
}
```

what should the config be for both of them ?

**Reply by juicybaba** [reply]
Hi @nabeelsaabna, please see the config for `TeckResourcesTDS`.
```
{
	"extends": [
		"local>TeckResourcesTDS/.github:renovate-config"
	],
	"onboarding": false,
	"requireConfig": "optional"
}
```

**Reply by nabeelsaabna** [reply]
done

**Reply by juicybaba** [reply]
looks good to me, thank you guys so much @nabeelsaabna @rarkins
